config/config.go

```diff
@@ -0,0 +1,8 @@
+package config
+
+// GenericConfig is an interface for all product based config types to implement
+type GenericConfig[T any] interface {
+	Validate() error
+	ApplyOverride(from T) error
```

Comment:
Should we add `ApplySecrets()` to the interface so we can put all secrets handling in these methods?

Reply:
I actually don't think we need to keep this interface.